### ISSUE TITLE
Not working in Android 12 (Require FLAG_MUTABLE for pending intent)

### ISSUE THREAD

**Issue by milan-galitein** [wonderpush/wonderpush-android-sdk#9]
My Application set targetSdkVersion 32 and when I run on Android 12 Device then notification not working.

Please check below document and update library.
https://developer.android.com/about/versions/12/behavior-changes-12#pending-intent-mutability

**Comment by ofavre** [comment]
Hi,

What do you see in logcat that makes you think PendingIntent mutability is the cause of the issue?

**Comment by milan-galitein** [comment]
Getting below error in LOG

  ```
2022-02-24 10:26:31.868 6222-7012/ E/WonderPush.Push.FCM.FirebaseMessagingService: Unexpected error while handling FCM message from:1023997258979 bundle: 

    java.lang.IllegalArgumentException: : Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
    Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
        at android.app.PendingIntent.checkFlags(PendingIntent.java:375)
        at android.app.PendingIntent.buildServicePendingIntent(PendingIntent.java:724)
        at android.app.PendingIntent.getService(PendingIntent.java:686)
        at com.wonderpush.sdk.NotificationManager$PendingIntentBuilder.buildForDataNotificationWillOpenBroadcast(NotificationManager.java:309)
        at com.wonderpush.sdk.NotificationManager.fetchResourcesAndDisplay(NotificationManager.java:196)
        at com.wonderpush.sdk.NotificationManager.onReceivedNotification(NotificationManager.java:134)
        at com.wonderpush.sdk.push.fcm.FirebaseMessagingService.onMessageReceived(FirebaseMessagingService.java:137)
        at com.wonderpush.sdk.push.fcm.FirebaseMessagingService.onMessageReceived(FirebaseMessagingService.java:98)
        at com.google.firebase.messaging.FirebaseMessagingService.dispatchMessage(com.google.firebase:firebase-messaging@@23.0.0:12)
        at com.google.firebase.messaging.FirebaseMessagingService.passMessageIntentToSdk(com.google.firebase:firebase-messaging@@23.0.0:8)
        at com.google.firebase.messaging.FirebaseMessagingService.handleMessageIntent(com.google.firebase:firebase-messaging@@23.0.0:3)
        at com.google.firebase.messaging.FirebaseMessagingService.handleIntent(com.google.firebase:firebase-messaging@@23.0.0:3)
        at com.google.firebase.messaging.EnhancedIntentService.lambda$processIntent$0$com-google-firebase-messaging-EnhancedIntentService(com.google.firebase:firebase-messaging@@23.0.0:1)
        at com.google.firebase.messaging.EnhancedIntentService$$ExternalSyntheticLambda1.run(Unknown Source:6)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at com.google.android.gms.common.util.concurrent.zza.run(com.google.android.gms:play-services-basement@@18.0.0:2)
        at java.lang.Thread.run(Thread.java:920)
```

**Comment by ofavre** [comment]
I believe you're not using an up-to-date version of our SDK, v4.0.6 or earlier probably.
This was fixed in v4.1.0. The latest version as of today is [v4.1.3](https://github.com/wonderpush/wonderpush-android-sdk/releases/tag/v4.1.3).
